python/tk_multi_workfiles/find_files.py

```diff
@@ -289,21 +292,30 @@ def __find_publishes(self, context, publish_template):
             
         return published_files
     
-        
-    def __find_work_files(self, context, work_template):
+    def __find_work_files(self, context, work_template, require_path):
         """
         Find all work files for the specified context and work template
-        
+
         :param context:             The context to find work files for
         :param publish_template:    The work template to match found files against
+        :param require_path:        If True, ensures that all fields of the template
+                                    can be resolved.
         :returns:                   List of dictionaries, each one containing the details
-                                    of an individual work file        
+                                    of an individual work file
+
+        :raises TankError: Raised when not all fields of the template can be resolved.
         """
         # find work files that match the current work template:
         work_fields = []
         try:
             work_fields = context.as_template_fields(work_template, validate=True)
         except TankError:
+            if require_path:
+                self.__app.log_exception("Unable to resolve all template fields.")
+                # and raise a new, clearer exception for this specific use case:
+                raise TankError("Unable to resolve template fields!  This could mean there is a mismatch "
```

Comment:
Good idea. I had reproduced the message from workfiles 2, but I think what you are right, we should show the proper error message. The error thrown by as_template_fields actually the context and template in it, so it's perfect.